Assets/MixedRealityToolkit.SDK/CompilationValidator/CompilationValidator.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.﻿
+
+using System;
+using System.Diagnostics;
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Utilities.Editor
+{
+    [InitializeOnLoad]
+    public class CompilationValidator
+    {
+        private const string warningTitle = "Assembly reference missing";
+        private const string warningMessage = "Unity 2019.2 and later requires that your UnityAR assembly definition be configured differently. We can attempt to do this automatically. Proceed?";
+        private const string unityArAsmdefPath = "Assets/MixedRealityToolkit.Staging/UnityAR/Microsoft.MixedReality.Toolkit.Providers.UnityAR.asmdef";
+        private const string spatialTrackingReference = "UnityEngine.SpatialTracking";
+        private const string powershellScriptPath = "scripts/support2019/setup_for_2019.ps1";
+
+        static CompilationValidator()
+        {
+#if UNITY_2019_2_OR_NEWER
+            TextAsset unityArAsmdef = AssetDatabase.LoadAssetAtPath<TextAsset>(unityArAsmdefPath);
+
+            if (!unityArAsmdef.text.Contains(spatialTrackingReference))
+            {
+                if (EditorUtility.DisplayDialog(warningTitle, warningMessage, "OK", "Cancel"))
+                {
+                    string path = Application.dataPath.Replace("Assets", powershellScriptPath);
```

Comment:
Note: This does not solve the issue for customers who use the asset packages from GitHub...

Reply:
@davidkline-ms Sorry, should have made it clear that this _only_ fixes the issue for people who pull from the github repo. My understanding was that we have alternate solutions in the works for other scenarios.

Reply:
we have a solution for the github repo (there is a script in the scripts folder already). anything that gets checked into one of the MixedRealityToolkit.* folders should address issues for customers using the unity or nuget packages (>90%)

Reply:
we should discuss a good way to call out the required steps for repository users, i 100% agree.

Reply:
@davidkline-ms  is there a way to add files that won't be shipped with the unitypackages/nuget?

Reply:
It depends on whether or not we want them to appear in the project window.... i will continue to work on this as it will be of minimal impact to the release schedule.

current thinking is to check the .asmdef contents in both 2018 and 2019 and display a dialog pointing the user to instructions, if not found. we can also do a similar check for the AR Foundation and XR Legacy packages.